subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeMultiProjectIntegrationTest.groovy

```diff
@@ -43,27 +43,202 @@ rootProject.name = "${WORKSPACE_NAME}"
+            ":hello:xcodeProject", ":hello:xcodeProjectWorkspaceSettings", ":hello:xcodeSchemehelloSharedLibrary", ":hello:xcode",
+            ":xcodeWorkspace", ":xcodeWorkspaceWorkspaceSettings", ":xcode")
+
+        xcodeWorkspace("${rootProjectName}.xcworkspace")
+            .contentFile.assertHasProjects([file("${rootProjectName}.xcodeproj"), file('app/app.xcodeproj'), file('greeting/greeting.xcodeproj'), file('hello/hello.xcodeproj')]*.absolutePath)
+
+        buildSettings(xcodeProject("app/app.xcodeproj").projectFile).SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file("hello/build/main/objs"), file("greeting/build/main/objs"))
+        buildSettings(xcodeProject("hello/hello.xcodeproj").projectFile).SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file("greeting/build/main/objs"))
+    }
+
+
+    def "create xcode project C++ executable"() {
+        given:
+        buildFile << """
+            project(':app') {
+                apply plugin: 'cpp-executable'
+                dependencies {
+                    implementation project(':greeter')
+                }
+            }
+            project(':greeter') {
+                apply plugin: 'cpp-library'
+            }
 """
-        app.library.sourceFiles.each { it.writeToFile(file("greeter/src/main/swift/$it.name")) }
-        app.executable.sourceFiles.each { it.writeToDir(file('app/src/main')) }
-        def mainFile = file('app/src/main/swift/main.swift')
-        mainFile.text = "import greeter\n\n${mainFile.text}"
+        def app = new CppHelloWorldApp()
+        app.library.writeSources(file('greeter/src/main'))
+        app.executable.writeSources(file('app/src/main'))
 
         when:
         succeeds("xcode")
 
-        then: 'tasks are executed as expected'
+        then:
         executedAndNotSkipped(":app:xcodeProject", ":app:xcodeProjectWorkspaceSettings", ":app:xcodeSchemeappExecutable", ":app:xcode",
             ":greeter:xcodeProject", ":greeter:xcodeProjectWorkspaceSettings", ":greeter:xcodeSchemegreeterSharedLibrary", ":greeter:xcode",
             ":xcodeWorkspace", ":xcodeWorkspaceWorkspaceSettings", ":xcode")
-        def workspaceXml = xcodeWorkspace("${WORKSPACE_NAME}.xcworkspace").contentFile.contentXml
 
-        and:
-        workspaceXml.FileRef.size() == 3
-        workspaceXml.FileRef*.@location*.replaceAll('absolute:', '').containsAll([file('app/app.xcodeproj'), file('greeter/greeter.xcodeproj')]*.absolutePath)
+        xcodeWorkspace("${rootProjectName}.xcworkspace")
+            .contentFile.assertHasProjects([file("${rootProjectName}.xcodeproj"), file('app/app.xcodeproj'), file('greeter/greeter.xcodeproj')]*.absolutePath)
+
+        buildSettings(xcodeProject("app/app.xcodeproj").projectFile).HEADER_SEARCH_PATHS == toSpaceSeparatedList(file("app/src/main/headers"), file("greeter/src/main/public"))
+    }
+
+    def "Gradle project with added xcode plugin are not included in the workspace"() {
```

Comment:
Oups, my bad. It's actually the opposite and that is what the test is actually doing. I will fix the test name to reflect what the test is actually testing.